gui/src/3D/controls/touchButton3D.ts

```diff
@@ -0,0 +1,387 @@
+// Assumptions: absolute position of button mesh is inside the mesh
+
+import { DeepImmutableObject } from "babylonjs/types";
+import { Vector3 } from "babylonjs/Maths/math.vector";
+import { Mesh } from "babylonjs/Meshes/mesh";
+import { AbstractMesh } from "babylonjs/Meshes/abstractMesh";
+import { TransformNode } from "babylonjs/Meshes/transformNode";
+import { Scene } from "babylonjs/scene";
+import { Ray } from "babylonjs/Culling/ray";
+
+import { Button3D } from "./button3D";
+
+/**
+ * Enum for Button States
+ */
+enum ButtonState {
+    /** None */
+    None = 0,
+    /** Pointer Entered */
+    Hover = 1,
+    /** Pointer Down */
+    Press = 2
+}
+
+class TouchButton3DManager {
+    private _touchButtonList = new Map<number, TouchButton3D>();
+    private _buttonIndex = 1;
+
+    private _sceneRegisteredOn: Scene | null;
+    private _collisionHandlingFunction: () => void;
+
+    /**
+     * Creates a new touchButton3DManager
+     */
+    constructor() {
+        let _this = this;
+        this._collisionHandlingFunction = function () {
+            if (_this._sceneRegisteredOn != null) {
+                const touchMeshes = _this._sceneRegisteredOn.getMeshesByTags("touchEnabled");
+
+                _this._touchButtonList.forEach(function (button: TouchButton3D) {
+                    touchMeshes.forEach(function (mesh: Mesh) {
+                        button.collisionCheckForStateChange(mesh);
+                    });
+                });
+            }
+        };
+    }
+
+    public addButton(button: TouchButton3D): number {
+        let index = this._buttonIndex++;

```

Comment:
>let [](start = 8, length = 3)

nit: it seems like this value doesn't change after construction. may want to use const instead